ffi/src/lib.rs

```diff
@@ -8,6 +8,66 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+//! This module exposes an Foreign Function Interface (FFI) that allows Mentat to be
+//! called from other languages.
+//!
+//! Functions that are available to other languages in this module are defined as
+//! extern "C" functions which allow them to be layed out correctly for the
+//! platform's C ABI. They all have a `#[no_mangle]` decorator to ensure
+//! Rust's name mangling is turned off, so that it is easier to link to.
+//!
+//! Mentat's FFI contains unsafe code. As it is an interface between foreign code
+//! and native Rust code, Rust cannot guarentee that the types and data that it is
+//! handling that have been passed to it from another language are present and in
+//! the format it is expecting.
+//! This interface is designed to ensure that nothing unsafe ever leaves this module
```

Comment:
nit: leaves?  This module is filtering unsafe inputs, correct?  Can we find a word that suggests that filtering -- perhaps, "nothing unsafe passes through this module and enters Mentat proper."?